src/pubsub/node_flat.erl

```diff
@@ -667,13 +583,11 @@ get_items(Nidx, _From, _RSM) ->
     {result, {lists:reverse(lists:keysort(#pubsub_item.modification, Items)), none}}.
 
 get_items(Nidx, JID, AccessModel, PresenceSubscription, RosterGroup, _SubId, RSM) ->
-    SubKey = jid:to_lower(JID),
-    GenKey = jid:to_bare(SubKey),
-    {ok, GenState} = mod_pubsub_db_backend:get_state(Nidx, GenKey),
-    {ok, SubState} = mod_pubsub_db_backend:get_state(Nidx, SubKey),
-    Affiliation = GenState#pubsub_state.affiliation,
-    BareSubscriptions = GenState#pubsub_state.subscriptions,
-    FullSubscriptions = SubState#pubsub_state.subscriptions,
+    {ok, Affiliation} = mod_pubsub_db_backend:get_affiliation(Nidx, JID),
```

Comment:
I would rather fail with something like #{reason => failed_to_get_affiliation, node_id => Nidx, jid => Jid} than badmatch without extra info. Just hope, that the error is super super rare and there is enough info added into error handling from the caller of the function.

Reply:
OK, I can come up with something. :)